src/org/labkey/targetedms/parser/SkylineBinaryParser.java

```diff
@@ -300,7 +300,23 @@ public int matchTransitions(ChromGroupHeaderInfo header, List<? extends GeneralT
             // For older data that got saved in the database without a value for uncompressedSize
             uncompressedSize = (Integer.SIZE / 8) * numPoints * (numTransitions + 1);
         }
-        return uncompress(bytes, uncompressedSize);
+        try
+        {
+            return uncompress(bytes, uncompressedSize);
+        }
+        catch (DataFormatException e)
+        {
+            // Issue 40469 - we may have -1 from a legacy import, but not have compressed data. If so, the attempt
+            // to uncompress will blow up and we can just return the original bytes
+            if (uncompressedSize.intValue() == -1)
```

Comment:
can we do this check before the return uncompress(bytes, uncompressedSize); ?

Reply:
Nope. I had hoped so too, but we don't know if it's compressed data or not.